src/core/qgsruntimeprofiler.cpp

```diff
@@ -575,6 +575,39 @@ QgsRuntimeProfilerNode *QgsRuntimeProfiler::index2node( const QModelIndex &index
   return reinterpret_cast<QgsRuntimeProfilerNode *>( index.internalPointer() );
 }
 
+void QgsRuntimeProfiler::extractModelAsText( QStringList &lines, const QString &group, const QModelIndex &parent, int level )
+{
+  const int rc = rowCount( parent );
+  const int cc = columnCount( parent );
+  for ( int r = 0; r < rc; r++ )
+  {
+    QModelIndex rowIndex = index( r, 0, parent );
+    if ( data( rowIndex, QgsRuntimeProfilerNode::Group ).toString() != group )
+      continue;
+
+    QStringList cells;
+    for ( int c = 0; c < cc; c++ )
+    {
+      QModelIndex cellIndex = index( r, c, parent );
+      cells << data( cellIndex ).toString();
+    }
+    lines << QStringLiteral( "%1 %2" ).arg( QStringLiteral( "-" ).repeated( level + 1 ), cells.join( QStringLiteral( ": " ) ) );
+    extractModelAsText( lines, group, rowIndex, level + 1 );
+  }
+}
+
+QString QgsRuntimeProfiler::asText()
+{
+  QStringList lines;
+  for ( const QString &group : std::as_const( mGroups ) )
```

Comment:
I'd suggest adding a QString "group" argument to this function, to allow the export of just a single profile group. It'd be rather tricky to filter the output to a single group afterwards if you're only after one profile.